tools/queries.cpp

```diff
@@ -176,14 +242,30 @@ void perftest(
 
     auto scorer = scorer::from_params(scorer_params, wdata);
 
-    spdlog::info("Performing {} queries", type);
-    spdlog::info("K: {}", k);
+    std::ofstream output_file;
+    std::ostream* output = nullptr;
+    if (output_path) {
+        output_file.open(*output_path);
+        if (!output_file.is_open()) {
+            const auto err_msg = fmt::format("Failed to open output file: {}.", *output_path);
+            spdlog::error(err_msg);
```

Comment:
I think rather than printing this error log here, we should make sure that `main()` function catches any exception and prints the error message.